samplelistener/listener/dispatcher_service.bal

```diff
@@ -2,46 +2,45 @@ import ballerina/http;
 import ballerina/jballerina.java;
 
 service class DispatcherService {
-   private SlackAppMentionHandlingService|SlackAppCreatedHandlingService serviceRef;
-   //private string verificationToken;
+     private map<GenericService> services;
 
-   isolated function init(SlackAppMentionHandlingService|SlackAppCreatedHandlingService serviceRef) returns error? { //Receive verification token
-        self.serviceRef = serviceRef;
-        //self.verificationToken = "9asdbas9009123nas1e2";
+   isolated function init() returns error? {
+        self.services = {};
    }
 
-   isolated resource function post events (http:Caller caller, http:Request request) returns error? {
-        json payload = check request.getJsonPayload();
-        // string eventOrVerification = check payload.'type;
-
-        // if (payload.token !== self.verificationToken) {
-        //     return error("Verification token mismatch");
-        // }
+   isolated function addServiceRef(string serviceType, GenericService genericService) {
+        self.services[serviceType] = genericService;
```

Comment:
Shouldn't we check if `self.services` already contains the key and return an error if so?

Reply:
Resolved